diff -urN src/shrip-common.c.orig src/shrip-common.c
--- src/shrip-common.c.orig	2008-07-06 05:09:14.000000000 -0700
+++ src/shrip-common.c	2009-04-09 10:30:57.000000000 -0700
@@ -1572,7 +1572,7 @@
   g_signal_connect (container, "run", G_CALLBACK (run), NULL);
 
   cwd = g_get_current_dir ();
-  chdir (ogmrip_fs_get_tmp_dir ());
+  char ignore = chdir (ogmrip_fs_get_tmp_dir ());
 
   spawn = OGMJOB_SPAWN (container);
   result = ogmjob_spawn_run (OGMJOB_SPAWN (container), &tmp_error);
@@ -1584,7 +1584,7 @@
   if (tmp_error)
     g_propagate_error (error, tmp_error);
 
-  chdir (cwd);
+  ignore = chdir (cwd);
   g_free (cwd);
 
   return result;
